dist/clouduploader.rb

```diff
@@ -71,7 +66,13 @@ def upload_image_to_ec2(image, data, jobid)
     "--session-token=#{credentials.session_token}",
     '--verbose',
     image
-  ) do |_stdin, stdout_stderr, _wait_thr|
+  ]
+
+  if data['vpc_subnet_id']
+    command << "--vpc-subnet-id=#{data['vpc_subnet_id']}"
```

Comment:
[nitpick] I fear that ec2uploadimg will not accept adding another argument after the image. But let's try ...